src/main/java/com/limechain/client/FullNode.java

```diff
@@ -103,14 +110,28 @@ public List<InsertTrieNode> buildTrieNodesIterator(
             if (userData.getValue1().isEmpty()) {
                 throw new IllegalStateException("Merkle value should not be empty!");
             }
-            NodeHandle<Pair<Optional<byte[]>, Optional<byte[]>>> nodeHandle = trieStructure.nodeHandleAtIndex(nodeIndex);
+            NodeHandle<Pair<Optional<byte[]>, Optional<byte[]>>> nodeHandle =
+                    trieStructure.nodeHandleAtIndex(nodeIndex);
 
-            byte[] storageValue = userData.getValue0().orElse(null);
+            Optional<byte[]> storageValueOpt = userData.getValue0();
             byte[] merkleValue = userData.getValue1().get();
+            final InsertTrieNode.InsertStorageValue storageValue = new InsertTrieNode.InsertStorageValue(
+                    storageValueOpt.orElse(null),
+                    storageValueOpt.isPresent(),
+                    false);
 
             byte[] merkleValueCopy = merkleValue.clone();
             List<byte[]> childrenMerkleValues = new ArrayList<>();
-            List<Nibble> partialKeyNibbles = new ArrayList<>(nodeHandle.getPartialKey().asUnmodifiableList());
+            Nibbles partialKey = nodeHandle.getPartialKey();
```

Comment:
You can use the `Bytes.toArray()` function from `com.google.common.primitives`, that's included, to make it more readable.
`byte[] partialKeyNibbles = Bytes.toArray(partialKey.asUnmodifiableList().stream().map(Nibble::asByte).toList());`